docs/src/doc/contribution/setup.md

```diff
@@ -1,36 +1,69 @@
+Once you have all the necessary dependencies, proceed to do the following:
 
-3. From the `godot-root` dir, build the engine with our module: `scons platform=linuxbsd # your target platform`
+1. Clone Godot's repository with the stable tag you want to develop for. Notice that the branch tag must be
+aligned to the current binding's version (e.g., current version `0.9.1-4.2.2`, we need Godot at `4.2.2` version).
 
-4. Build sample
-    - navigate to `<module-root>/harness/tests`
-    - create embedded JVM:
-        - For amd64 systems:
+```bash
+git clone git@github.com:godotengine/godot.git --branch 4.2.2-stable --recursive
+```
+
+2. In the `godot` directory, run the following command:
+```bash
+git submodule add git@github.com:utopia-rise/godot-kotlin-jvm.git modules/kotlin_jvm
+```
+
+3. Once the [git submodule](https://git-scm.com/book/en/v2/Git-Tools-Submodules) is added, change your directory to `godot`, and build the engine with our module:
+```bash
+scons platform=linuxbsd # your target platform (windows, macos, ...)
+```
+
+4. Build the sample following these steps:
+    1. Navigate to `<module-root>/harness/tests`
+    2. Create an embedded Java Virtual Machine:
+        - For `amd64` systems:
           - `jlink --add-modules java.base,java.logging --output jre-amd64`
-        - For arm64 systems:
+        - For `arm64` (macOS with **Apple Silicon** chips) systems:
           - `jlink --add-modules java.base,java.logging --output jre-arm64`
         - If you want to remote debug add module `jdk.jdwp.agent` to command.
-        - If you want to enable jmx, add `jdk.management.agent` to command.
-    - Windows: `gradlew build`
-    - Unix: `./gradlew build`
+        - If you want to enable `jmx`, add `jdk.management.agent` to command.
+    3. Then build the project using the Gradle wrapper.
+        - Windows: `gradlew build`
+        - Unix: `./gradlew build`
 
-5. In order to run the engine, run `godot.linuxbsd.editor.x86_64` (or the equivalent of your platform) located in the `bin` folder of `godot-root`
+5. To run the engine, run `godot.linuxbsd.editor.x86_64` (or the equivalent of your platform) located in the `bin` folder of `godot`.
 
-6. In order to debug your JVM code, you should start Godot with command line `--jvm-debug-port=XXXX`, where `XXXX`
-stands for the JMX port of you choice. You can then set up remote debug configuration in Intellij IDEA.
+6. To debug your JVM code, you should start Godot with command line `--jvm-debug-port=XXXX`, where `XXXX`
+stands for the JMX port of you choice.
+You can then set up [remote debug configuration in Intellij IDEA](https://www.jetbrains.com/help/idea/tutorial-remote-debug.html).
 
 
 ## Publishing locally
 
-In order to publish our artifacts locally, you'll need to run `gradlew :tools-common:publishToMavenLocal publishToMavenLocal`
+To publish our artifacts locally, you'll need to run the following command:
+```bash
+./gradlew :tools-common:publishToMavenLocal publishToMavenLocal
```

Comment:
Actually it's `./gradlew :tools-common:publishToMavenLocal publishToMavenLocal && ./gradlew publishToMavenLocal -Prelease=true` by now as you need both the library-debug and the library-release